### PR TITLE
fix multi nccl comm overlap

### PR DESCRIPTION
**Pull request by wangxicoding**
### PR types

Performance optimization
### PR changes

Others
### Describe

fix multi nccl comm overlap.
Before this PR, multi comm not overlap
image
After this PR, multi comm can overlap
image

TODO. Add performace.

### CODE REVIEW

File: paddle/fluid/framework/ir/multi_devices_graph_pass/all_reduce_deps_pass.cc [hunk 1]
```diff
@@ -48,15 +50,29 @@ class AllReduceDepsPass : public ir::Pass {
...
+    interval = nccl_comm_num;
 #endif
 
-    for (size_t i = 1; i < all_reduce_op_handles.size(); ++i) {
+    for (size_t i = interval; i < all_reduce_op_handles.size(); ++i) {
       auto* dep_var = new details::DummyVarHandle(graph->CreateControlDepVar());
       graph->Get<details::GraphDepVars>(details::kGraphDepVars)
           .emplace(dep_var);
-      all_reduce_op_handles[i - 1]->AddOutput(dep_var);
+      all_reduce_op_handles[i - interval]->AddOutput(dep_var);
```
Comment (gongweibao) [inline paddle/fluid/framework/ir/multi_devices_graph_pass/all_reduce_deps_pass.cc:75]: 赞

Comment (gongweibao) [reply paddle/fluid/framework/ir/multi_devices_graph_pass/all_reduce_deps_pass.cc:75]: 只不过有个问题，这样第0层的allreduce会不会乱序执行？

**Review by gongweibao — APPROVED** [review]
LGTM

**Comment by wangxicoding** [comment]
Close, may hang, use this PR https://github.com/PaddlePaddle/Paddle/pull/28663

**Pull request closed without merging**